### PR TITLE
Specify 'PASSWORD_STORE_DIR' when invoking pass from qute-pass

### PR DESCRIPTION
**Pull request by gnofs**
The `--password-store` argument to `qute-pass` does not affect the invocation of `pass` in the `_run_pass` function.
This leads to a situation where the password is found to exist and the username is returned, but no password is returned (unless the password is found in the default password store).

Easily solved by propagating the relevant path to `pass` as the environment variable `PASSWORD_STORE_DIR`. At least with pass v1.7.3, this is the only documented way of using a non-default password store.

### CODE REVIEW

File: misc/userscripts/qute-pass
Comment (The-Compiler) [inline misc/userscripts/qute-pass:140]: Setting `env=` here will mean that any existing environment variables will be cleared - thus, things like `XDG_*_HOME`, `EDITOR`, `HOME`, exiting `PASSWORD_STORE_*` variables, ... all won't go to pass anymore.

Perhaps it'd be better to do:

```python3
env = os.environ.copy()
env['PASSWORD_STORE_DIR'] = password_store
subprocess.run(..., env=env)
```

instead?

**Review by The-Compiler — CHANGES_REQUESTED** [review]

**Comment by The-Compiler** [comment]
cc @bascht @lufte if you have the time to take a quick look as well

File: misc/userscripts/qute-pass
Comment (gnofs) [reply misc/userscripts/qute-pass]: Good point, I agree.

File: misc/userscripts/qute-pass
Comment (bascht) [inline misc/userscripts/qute-pass:141]: I haven't tested it  locally, but I think you can also access `arguments.password_store` directly, instead of passing it via `pass_()` and `pass_otp()`, no?

**Review by bascht — CHANGES_REQUESTED** [review]
It'd be great if you could try to access the arguments directly – it should be in the scope like `arguments.mode` or `arguments.username_target`.

File: misc/userscripts/qute-pass
Comment (The-Compiler) [reply misc/userscripts/qute-pass:141]: Given that we already pass `encoding` as commandline argument as well, I think it makes more sense to be consistent here.

Comment (gnofs) [reply misc/userscripts/qute-pass:141]: @bascht: You're right though, arguments is defined outside the main scope, so it would be possible. We could alternatively replace both `encoding` and `password_store` with `arguments` attributes.

It would also be possible to use the parsed/expanded variable `password_store_path`. Shouldn't make any difference in this case, but it would be more consistent to use the exact same string for all use cases. This won't be possible if `arguments` is used, unless of course it is modified.

Comment (The-Compiler) [reply misc/userscripts/qute-pass:141]: Good point, I think indeed using `password_store_path` would be a good idea. Right now, if someone spawned `qute-pass` with `--password-store ~/some/path` from qutebrowser, things most likely still wouldn't work properly, because the `~` wouldn't get expanded in the environment variable.

**Review by The-Compiler — APPROVED** [review]
I guess that works as well - thanks for the quick follow-ups! :+1: Will give @bascht / @lufte a bit more time to have a look and merge this in a couple of days if nothing else comes up.

File: misc/userscripts/qute-pass
Comment (gnofs) [reply misc/userscripts/qute-pass]: I made a commit now in which the password store path is expanded in the argument parsing process, regardless of its source, so that both `arguments.io_encoding` and `arguments.password_store` can be accessed directly when needed. Less duplication of concerns, and more consistent code.

**Comment by lufte** [comment]
Good catch @gnofs. I guess you are the first one to ever user this option.

I have tested it and found no issues. On the other hand, why are we having some of these functions read arguments from a global variable instead of function parameters? It's not related to what's being fixed, and (in my opinion) it's not an improvement. Anyway, I see that you guys already discussed this and it was explicitly requested, so I'll leave it to you.

Great work @gnofs :+1:

**Comment by bascht** [comment]
@gnofs I second that, but this might also bring us dangerously close to "rewrite" territory. :grimacing: 

So far I like @gnofs's cleanup, as it already brings a lot more clarity. :+1:

**Comment by The-Compiler** [comment]
I guess there are arguments (no pun intended) for and against having everything as function arguments vs. globally - for a simple script like this I'm fine with having globals instead of having to pass a lot of things around everywhere.

Thanks @gnofs for the PR and @bascht @lufte for the reviews! :+1:

**Pull request merged**
